packages/adp-tooling/src/preview/change-handler.ts

```diff
@@ -0,0 +1,73 @@
+import type { Editor } from 'mem-fs-editor';
+import { TemplateFileName, FolderTypes } from '../types';
+import type { AddXMLChange, CommonChangeProperties, CodeExtChange } from '../types';
+import { join } from 'path';
+import type { Logger } from '@sap-ux/logger';
+
+/**
+ * A mapping object that defines how to extract change content data from changes based on their type.
+ */
+export const moduleNameContentMap: { [key: string]: (change: CommonChangeProperties) => string } = {
+    codeExt: (change) => ((change as CodeExtChange).content?.codeRef ?? '').replace('.js', ''),
+    addXML: (change) => (change as AddXMLChange).content?.fragmentPath ?? ''
+} as const;
```

Comment:
Are we sure that change always has content?

Reply:
For both the `codeExt` and `addXML` functions within `moduleNameContentMap`, if `change.content` is `undefined`, the outcome will be an empty string (`''`). The optional chaining operator (`?.`) prevents a runtime error and returns `undefined`. If `undefined` is returned then `??` returns an empty string `''`. Then when `.replace('.js', '')` is called on the empty string, effectively leaving it **unchanged**.